brightcove-exoplayer/OfflinePlaybackSampleApp/src/main/java/com/brightcove/player/samples/offlineplayback/VideoListAdapter.java

```diff
@@ -242,13 +262,36 @@ public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
      *
      * @param holder   reference to the view holder.
      * @param position the position of the row that should be updated.
-     * @param status   optional current download status.
+     * @param downloadStatus   optional current download status.
      */
-    private void updateView(@NonNull final ViewHolder holder, int position, @Nullable DownloadStatus status) {
+    private void updateView(@NonNull final ViewHolder holder,
+                            int position,
+                            @Nullable final DownloadStatus downloadStatus) {
         holder.video = videoList.get(position);
-        final Video video = holder.getVideo();
+        holder.getVideo(new OfflineCallback<Video>() {
+            @Override
+            public void onSuccess(Video video) {
+                handleViewState(holder, video, downloadStatus);
+            }
+
+            @Override
+            public void onFailure(Throwable throwable) {
+                Log.e(TAG, "Error fetching video: ", throwable);
+            }
+        });
+    }
 
+    private void handleViewState(final ViewHolder holder, Video video, DownloadStatus status) {
         if (video.isOfflinePlaybackAllowed()) {
+            holder.estimatedSizeText.setVisibility(View.VISIBLE);
+            catalog.estimateSize(video, new MediaDownloadable.OnVideoSizeCallback() {
+                @Override
+                public void onVideoSizeEstimated(long l) {
+                    String sizeStr = String.valueOf(l / 1048576).concat(" MB");
```

Comment:
Ideally, we should always use constants variables, that helps with readability, for example:
`static final long MEGABYTES_IN_BYTES = 1048576`